src/array/first.ts

```diff
@@ -11,9 +11,11 @@
  * // 0
  * ```
  */
-export function first<T>(array: readonly T[]): T | undefined
+export function first(array: readonly []): undefined
+export function first<T>(array: readonly T[]): T
```

Comment:
Can't assume the length of the array type is not zero here, so return type has to be `T | undefined`. Or is your goal to align this with `arr[0]`-style access?

Reply:
I think it's the same case as https://github.com/radashi-org/radashi/pull/139#issuecomment-2254329876

Reply:
Agreed, invalid for similar reasons. Suggest closing.